packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1231,80 +1384,13 @@ class TextInput {
     return true;
   }
 
-  late MethodChannel _channel;
-
+  final _TextInputSource _currentSource = _TextInputSource();
```

Comment:
This class will be made public in a different PR?

Reply:
The next step is to make this an abstract public class and move the default implementation into a private subclass. We have most of it already covered in the original PR #69146 but especially the docs need some work. I will rebase #69146 and continue where we left off.